### PR TITLE
bump to react-18

### PR DESCRIPTION
**Pull request by BSd3v**
- fix of `openGroups`
- fix `paginationGoTo`
- fix `rowMenuRenderer` `material-ui` to most recent version

### CODE REVIEW

File: src/lib/fragments/AgGrid.react.js [hunk 1]
```diff
@@ -648,93 +732,16 @@ export default class DashAgGrid extends Component {
...
-        } = this.props;
+        const {rowModelType} = this.props;
 
-        const propsToSet = {};
         if (rowModelType === 'infinite') {
             params.api.setDatasource(this.getDatasource());
         }
 
-        this.setState({
+        this.setState(() => ({
```
Comment (alexcjohnson) [inline src/lib/fragments/AgGrid.react.js:741]: Is it necessary to make this a function?

Comment (BSd3v) [reply src/lib/fragments/AgGrid.react.js:741]: No, was trying to solve the bad `setState` issue, we can revert.

File: package.json [hunk 2]
```diff
@@ -28,8 +28,10 @@
     "author": "Plotly <chris@plot.ly>",
     "license": "MIT",
     "dependencies": {
-        "@material-ui/core": "^4.11.4",
-        "@material-ui/icons": "^4.11.3",
+        "@mui/material": "^5.12.1",
+        "@mui/icons-material": "^5.11.16",
+        "@emotion/styled": "^11.10.6",
+        "@emotion/react": "^11.10.6",
```
Comment (alexcjohnson) [inline package.json:34]: Why are these `@emotion` packages here?

Comment (BSd3v) [reply package.json:34]: Dependency of `@mui/material`

Comment (alexcjohnson) [reply package.json:34]: I see, @BSd3v pointed me to this odd aspect of their [installation instructions](https://mui.com/material-ui/getting-started/installation/) that you need to explicitly install those even if you don't use them directly 🙄

Comment (AnnMarieW) [reply package.json:34]: Is there any way to just include the row menu component rather than the entire package?

**Pull request merged**